File: rust/src/golem.rs
pub struct Golem
{
    pub hp: i32,
    pub defense: i32,
    pub attack: i32
}

impl Golem {
    pub fn new(hp: i32) -> Golem {
        let mut _golem = Golem::default();
        _golem.hp = hp;
        _golem
    }
}

impl Default for Golem {
    fn default() -> Self {
        Golem {
            hp: 300,
            defense: 80,
            attack: 50
        }
    }
}

pub struct GolemEx {
    pub hp: i32,
    pub defense: i32,
    pub attack: i32,
    pub hit: i32,
    pub flee: i32
}
impl GolemEx {
    const GOLEM_HIT: i32 = 70;
    const GOLEM_FLEE: i32 = 20;
    pub fn new(hp: i32, defense: i32, attack: i32) -> GolemEx {
        let mut golem = GolemEx {
            hp,
            defense,
            attack,
            hit: GolemEx::GOLEM_HIT,
            flee: GolemEx::GOLEM_FLEE
        };
        golem
    }
}

File: Dart/lib/Golem.dart
const int GOLEM_HIT = 70;
const int GOLEM_FLEE = 20;

class Golem {
  int Hp;
  int Defense;
  int Attack;

  Golem({int hp = 300, int defense = 80, int attack = 50}) {
    this.Hp = hp;
    this.Defense = defense;
    this.Attack = attack;
  }
}

class GolemEx {
  int Hp;
  int Defense;
  int Attack;
  int Hit;
  int Flee;

  GolemEx(int hp, int defense, int attack) {
    this.Hp = hp;
    this.Defense = defense;
    this.Attack = attack;
    this.Hit = GOLEM_HIT;
    this.Flee = GOLEM_FLEE;
  }
}